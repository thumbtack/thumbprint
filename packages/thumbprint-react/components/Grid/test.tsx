import React from 'react';
import { mount } from 'enzyme';
import { Grid, GridColumn } from './index';

test('grid renders content that is passed in', () => {
    const wrapper = mount(<Grid>goose</Grid>);
    expect(wrapper.text()).toBe('goose');
    expect(wrapper).toMatchSnapshot();
});

test('grid renders with wide gutter', () => {
    const wrapper = mount(<Grid gutter="wide">goose</Grid>);
    expect(wrapper.text()).toBe('goose');
    expect(wrapper).toMatchSnapshot();
});

test('grid renders with flush gutter', () => {
    const wrapper = mount(<Grid gutter="flush">goose</Grid>);
    expect(wrapper.text()).toBe('goose');
    expect(wrapper).toMatchSnapshot();
});

describe('GridColumn', () => {
    test('column renders content that is passed in', () => {
        const wrapper = mount(
            <Grid>
                <GridColumn>goose</GridColumn>
            </Grid>,
        );
        expect(wrapper.text()).toBe('goose');
        expect(wrapper).toMatchSnapshot();
    });

    test('column renders content with custom widths', () => {
        const wrapper = mount(
            <Grid>
                <GridColumn base={1} aboveSmall={2} aboveMedium={3} aboveLarge={4}>
                    goose
                </GridColumn>
            </Grid>,
        );
        expect(wrapper.text()).toBe('goose');
        expect(wrapper).toMatchSnapshot();
    });

    test('adds `dataTest` prop', () => {
        const wrapper = mount(
            <Grid dataTest="Duck">
                <GridColumn base={1} dataTest="Goose">
                    goose
                </GridColumn>
            </Grid>,
        );
        expect(wrapper.find('div[data-test="Duck"]')).toHaveLength(1);
        expect(wrapper.find('div[data-test="Goose"]')).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    test('adds `dataTestId` prop', () => {
        const wrapper = mount(
            <Grid dataTestId="Duck">
                <GridColumn base={1} dataTestId="Goose">
                    goose
                </GridColumn>
            </Grid>,
        );
        expect(wrapper.find('div[data-testid="Duck"]')).toHaveLength(1);
        expect(wrapper.find('div[data-testid="Goose"]')).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });
});
