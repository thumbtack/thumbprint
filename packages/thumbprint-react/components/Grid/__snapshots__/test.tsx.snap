// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridColumn adds \`dataTest\` prop 1`] = `
<Grid
  dataTest="Duck"
>
  <div
    className="grid"
    data-test="Duck"
  >
    <GridColumn
      base={1}
      dataTest="Goose"
    >
      <div
        className="col col1"
        data-test="Goose"
      >
        goose
      </div>
    </GridColumn>
  </div>
</Grid>
`;

exports[`GridColumn adds \`dataTestId\` prop 1`] = `
<Grid
  dataTestId="Duck"
>
  <div
    className="grid"
    data-testid="Duck"
  >
    <GridColumn
      base={1}
      dataTestId="Goose"
    >
      <div
        className="col col1"
        data-testid="Goose"
      >
        goose
      </div>
    </GridColumn>
  </div>
</Grid>
`;

exports[`GridColumn column renders content that is passed in 1`] = `
<Grid>
  <div
    className="grid"
  >
    <GridColumn>
      <div
        className="col col12"
      >
        goose
      </div>
    </GridColumn>
  </div>
</Grid>
`;

exports[`GridColumn column renders content with custom widths 1`] = `
<Grid>
  <div
    className="grid"
  >
    <GridColumn
      aboveLarge={4}
      aboveMedium={3}
      aboveSmall={2}
      base={1}
    >
      <div
        className="col col1 aboveSmallCol2 aboveMediumCol3 aboveLargeCol4"
      >
        goose
      </div>
    </GridColumn>
  </div>
</Grid>
`;

exports[`grid renders content that is passed in 1`] = `
<Grid>
  <div
    className="grid"
  >
    goose
  </div>
</Grid>
`;

exports[`grid renders with flush gutter 1`] = `
<Grid
  gutter="flush"
>
  <div
    className="grid gridFlush"
  >
    goose
  </div>
</Grid>
`;

exports[`grid renders with wide gutter 1`] = `
<Grid
  gutter="wide"
>
  <div
    className="grid gridWide"
  >
    goose
  </div>
</Grid>
`;
